src/pkg/pb/terrariumpb/service.proto

```diff
@@ -114,3 +114,16 @@ service TerrariumService {
     };
   }
 }
+
+message Dependency {
+  string interface_id = 1;
+  string title = 2;
+  string description = 3;
+  string inputs = 4; // JSON serialized Inputs
+  string outputs = 5; // JSON serialized Outputs
```

Comment:
instead of json string, this needs to be more structured for @shubhamseervi. we have done something similar in vg-api: https://github.com/cldcvr/vanguard-api/blob/master/pkg/pb/common/jsonschema.proto

something like:
```proto
message Dependency {
  string interface_id = 1;
  string title = 2;
  string description = 3;
  JSONSchema inputs = 4; // JSON serialized Inputs
  JSONSchema outputs = 5; // JSON serialized Outputs
}

message JSONSchema { // ref: http://forivall.com/json-schema-cheatsheet/
  string title = 1;
  string description = 2;
  string type = 3; // string, number, integer, boolean, object, array, null
  google.protobuf.Value default = 4; // the default value

  // string + number + integer
  repeated google.protobuf.Value examples = 5;
  repeated google.protobuf.Value enum = 6;

  // string. ref: https://json-schema.org/understanding-json-schema/reference/string.html
  int32 minLength = 10;
  int32 maxLength = 11;
  string pattern = 12;
  string format = 13;

  // number + integer. ref: https://json-schema.org/understanding-json-schema/reference/numeric.html
  int32 minimum = 20;
  int32 maximum = 21;
  int32 exclusiveMinimum = 22;
  int32 exclusiveMaximum = 24;
  int32 multipleOf = 25;

  // array. ref: https://json-schema.org/understanding-json-schema/reference/array.html
  JSONSchema items = 30;
  bool additionalItems = 31;
  int32 minItems = 32;
  int32 maxItems = 33;
  bool uniqueItems = 34;

  // object. ref: https://json-schema.org/understanding-json-schema/reference/object.html
  map<string, JSONSchema> properties = 40;
  repeated string required = 41;
}
```